src/content/url-handling.js

```diff
@@ -35,6 +35,8 @@ function dataSourceDirs(urlState: URLState) {
       return ['local', urlState.hash];
     case 'public':
       return ['public', urlState.hash];
+    case 'from-url':
+      return ['from-url', urlState.hash];
```

Comment:
Please add a separate field for this instead of re-using `hash`. You could call it `profileURL`.